src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -363,107 +366,133 @@ private void LogCachingLogger(TContext rootContext, TContext context, bool clone
-                }
-                else
-                {
-                    directories.Enqueue(directory);
-                }
+                    if (options.Recurse)
+                    {
+                        EnqueueAllDirectories(directories, directory);
+                    }
+                    else
+                    {
+                        directories.Enqueue(directory);
+                    }
 
-                var sortedFiles = new SortedSet<string>();
+                    var sortedFiles = new SortedSet<string>();
 
-                while (directories.Count > 0)
-                {
-                    sortedFiles.Clear();
+                    while (directories.Count > 0)
+                    {
+                        sortedFiles.Clear();
 
-                    directory = Path.GetFullPath(directories.Dequeue());
+                        directory = Path.GetFullPath(directories.Dequeue());
 
 #if NETFRAMEWORK
-                    // .NET Framework: Directory.Enumerate with empty filter returns NO files.
-                    // .NET Core: Directory.Enumerate with empty filter returns all files in directory.
-                    // We will standardize on the .NET Core behavior.
-                    if (string.IsNullOrEmpty(filter))
-                    {
-                        filter = "*";
-                    }
+                        // .NET Framework: Directory.Enumerate with empty filter returns NO files.
+                        // .NET Core: Directory.Enumerate with empty filter returns all files in directory.
+                        // We will standardize on the .NET Core behavior.
+                        if (string.IsNullOrEmpty(filter))
+                        {
+                            filter = "*";
+                        }
 #endif
 
-                    foreach (string file in FileSystem.DirectoryEnumerateFiles(directory, filter, SearchOption.TopDirectoryOnly))
-                    {
-                        // Only include files that are below the max size limit.
-                        if (IsTargetWithinFileSizeLimit(file, _rootContext.MaxFileSizeInKilobytes))
+                        foreach (string file in FileSystem.DirectoryEnumerateFiles(directory, filter, SearchOption.TopDirectoryOnly))
                         {
-                            sortedFiles.Add(file);
-                            continue;
+                            // Only include files that are below the max size limit.
+                            if (ShouldEnqueue(file, _rootContext))
+                            {
+                                sortedFiles.Add(file);
+                                continue;
+                            }
+
+                            if (!IsTargetWithinFileSizeLimit(file, _rootContext.MaxFileSizeInKilobytes, out long fileSizeInKb))
+                            { 
+                                _ignoredFilesCount++;
+                            }
```

Comment:
question, 
we use the ShouldEnqueue() to see if to add to sortedFiles list.
we use the IsTargetWithinFileSizeLimit to see if to add to _ignoredFilesCount.

Should we use the same logic, by just using the `else`

We can add more logics to `ShouldEnqueue` in the future.